### ISSUE TITLE
Haskell highlighting doesn't support multi-function type declarations

### ISSUE THREAD

**Issue by taksuyu** [zedapp/zed#479]
When given something like

``` haskell
fun1, fun2, funN :: Int -> Int
```

Zed doesn't properly notice it's a type declaration.

Specifically it doesn't highlight the function names but it gets the actual declaration.

**Comment by iElectric** [comment]
That's upstream problem of ACE, see https://github.com/zedapp/zed/blob/master/app/config/default/mode/haskell.json

**Comment by zefhemel** [comment]
Indeed, if this problem is fixed in ACE we can pull it in here. Please report it here: https://github.com/ajaxorg/ace
